ice-adapter/src/main/java/com/faforever/iceadapter/debug/TelemetryDebugger.java

```diff
@@ -87,10 +74,12 @@ public void onError(Exception ex) {
         objectMapper = new ObjectMapper();
         objectMapper.registerModule(new JavaTimeModule());
 
-        sendingLoopThread = new Thread(this::sendingLoop, "sendingLoop");
-        sendingLoopThread.setUncaughtExceptionHandler(
-                (t, e) -> log.error("Thread sendingLoop crashed unexpectedly", e));
-        sendingLoopThread.start();
+        CompletableFuture.runAsync(
+                () -> {
+                    Thread.currentThread().setName("sendingLoop");
+                    sendingLoop();
+                },
+                executor);
```

Comment:
Feels dangerous. If you don't pass a OneNewVirtualThreadPerTask executor, you might rename and block a shared pool thread.

Since the thread is a loop and not a one time task, why don't just spawn a virtual thread here and leave out the completable future?